### PR TITLE
Add derivative integration

### PR DESCRIPTION
**Pull request by afaucogney**
**Description:**

Add derivative component

**Pull request in home-assistant (if applicable):** home-assistant/home-assistant#26456

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next Home Assistant release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_components/derivative.markdown [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+title: "Derivative Sensor"
+description: "Instructions on how to integrate Derivative Sensor into Home Assistant."
+ha_category:
+  - Utility
+  - Energy
+ha_release: 0.99
+ha_iot_class: Local Push
+logo: derivative.png
+ha_qa_scale: internal
+redirect_from:
```
Comment (klaasnicolaas) [inline source/_components/derivative.markdown]: Please remove the `redirect_from`

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

File: source/_components/derivative.markdown (hunk 1 above)
Comment (afaucogney) [reply source/_components/derivative.markdown]: @klaasnicolaas Could tell me the reason (first pr for me)

Comment (klaasnicolaas) [reply source/_components/derivative.markdown]: We used to have separate pages for an integration for each platform that it supported, in the end it was decided to merge all platform pages into 1 integration page (with a few exceptions).

We now only use the `redirect_from` to refer existing integrations to a new filename format (`sensor.test.markdown` to `test.markdown`) and therefore does not apply to new integrations.

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.
⚠️ The `ha_release` might need an update before merging this one in.

**Pull request merged**